[Editor] A new CurrentPointers class to store current pointers used by the editor

legraina opened the pull request:
I have created a different PR as requested.

Move current pointer field of DrawingEditor to CurrentPointer class in tools.js. Also, only reset pointer type when user select a new mode.

- [x] PointerType: 
  - [x] The pointer types fields have been moved to a CurrentPointer object in tools.js. This object is used by eraser.js and ink.js 
  - [x] Clear the pointer type when changing mode, instead of at the end of the session. It seems more stable, as the method is not called this way when the user changes pages.
  - [x] Only clear the pointer type when the mode is changed by an event (the user changes the editor type), otherwise, the same pointer type is kept (the document is changed for example).

calixteman commented on src/display/editor/tools.js:
```diff
@@ -42,6 +42,78 @@ function bindEvents(obj, element, names) {
+    return false;
+  }
+
+  static isPointerType() {
```
I'm not an expert in english, but the name looks strange for me... I'd have said `hasPointerType`.
I just read again https://github.com/mozilla/pdf.js/pull/20213/files#diff-02919db67aef2f378464ba4c97e5f59e1e114498a147113025cd4163287f25f4R670 and I've the feeling that this class should be initialized with this the pointer type and the static methods in this class shouldn't be static. An instance could be created in startDrawing and set in a static member of DrawingEditor.

legraina replied:
I've changed the name.
For the logic, I feel that we need a static class outside of the editor if we want all editors to share the same pointer type. It's really important for pdf with multiple pages as each page has its own editor, and you don't want to reinitialized the pointer type on each page. Furthermore, when you use this mode for the first time, you want to initialize the pointer type, and the only way to do that is to check if the pointer type is null. Also, you will have to put the class with a getter/setter if you want to be able to clear it outside of the DrawingEditor (it's the only way to check if the mode is changed to the best of my knowledge, or perhaps through ``disableEditing``).

legraina replied:
Thanks for the answer. Even if you are okay with that, I want to highlight the fact that I don't think it's just an eraser issue. Imagine that you are drawing on one page with a stylus, then you scroll with your finger to the next page, you use your finger to stop the scroll, and finally you start drawing: I'm not convinced that the pointer type will be set on stylus and not touch when you try drawing.

legraina committed:
A new CurrentPointers class to store current pointers used by the editor

Move current pointer field of DrawingEditor to CurrentPointer class in tools.js: The pointer types fields have been moved to a CurrentPointer object in tools.js. This object is used by eraser.js and ink.js.
Only reset pointer type when user select a new mode: Clear the pointer type when changing mode, instead of at the end of the session. It seems more stable, as the method is not called this way when the user changes pages. Also, clear the pointer type when the mode is changed by an event (the user changes the editor type), otherwise, the same pointer type is kept (the document is changed for example)

legraina commented on src/display/editor/tools.js:
```diff
@@ -42,6 +42,78 @@ function bindEvents(obj, element, names) {
+      return true;
+    }
+
+    CurrentPointer.#pointerIds?.delete(pointerId);
```
I've changed the function name to ``isSamePointerIdOrRemove``

legraina commented on src/display/editor/tools.js:
```diff
@@ -42,6 +42,78 @@ function bindEvents(obj, element, names) {
 }
 
+/**
+ * Class to store current pointer used by the editor.
```
I've updated the comment, and change the name of the class to CurrentPointers.

legraina commented on src/display/editor/draw.js:
```diff
@@ -684,10 +677,7 @@ class DrawingEditor extends AnnotationEditor {
-      DrawingEditor.#currentPointerType &&
-      DrawingEditor.#currentPointerType !== pointerType
-    ) {
+    if (CurrentPointer.isInitializedAndDifferentPointerType(pointerType)) {
```
It's done.

legraina reviewed (commented):
I've made most of the modifications asked.

legraina commented:
> @legraina could you rebase and squash your PR please ?

@calixteman Done

timvandermeij commented:
Please fix the commit message to have a descriptive title and body so that when reading the `git log` output it's easy to tell what the change was and why it was introduced.

In general I can recommend https://cbea.ms/git-commit/ which explains how to write good commit messages and why that matters.

calixteman reviewed (approved):
LGTM. Thank you.
